### ISSUE TITLE
Expand `~` to home path when using double quotes

### ISSUE THREAD

**Issue by EdenEast** [nushell/nushell#5882]
### Describe the bug

```nu
> "~/.local/share/nushell/login-env.nu" | path exists
false

> "/home/user/.local/share/nushell/login-env.nu" | path exists
true
```

I am following the section [Configuring nu as a login shell][doc] from the docs. There is creates a list of `let-env`
expressions that I am supposed to add to my `env.nu` file. Since this information is local to this machine I have saved
the output into `~/.local/share/nushell/login-env.nu`. In my `~/.config/nushell/env.nu` I want to source this file if it
exists at the beginning of the script. As I commit my `~/.config` folder to a git repository and use it on multiple
machines I need to source the local file.

Changing to constructing the path does not work either as the `source` command does not understand variables. (Relates
to #2990)

```nu
let local-env-path  = ([$nu.home-path local share nushell login-env.nu] | path join)
if ($local-env-path | path exists) {
    source $local-env-path
}
```

```console
 3 │     source $local-env-path
   ·            ───────┬───────
   ·                   ╰── File not found: $local-env-path
```

[doc]: https://www.nushell.sh/book/configuration.html#configuring-nu-as-a-login-shell

### How to reproduce

1. Use path with `~` as the home root to a file that exists and see that `path exists` returns `false`
2. Use the full path to that same file and see that `path exists` returns `true`

### Expected behavior

I expected that `~` character would be expanded when using `""` quotes when using the path command.

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.63.0                                          |
| branch             |                                                 |
| commit_hash        |                                                 |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.61.0                                    |
| cargo_version      | cargo 1.61.0                                    |
| pkg_version        | 0.63.0                                          |
| build_time         | 1980-01-01 00:00:00 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  |                                                 |

**Comment by kubouch** [comment]
`"~/.local/share/nushell/login-env.nu" | path exists` should say `true`, that's a bug in the command.

Btw, it's not possible to source files from variables and also Nushell is scoped so it the source won't have any effect outside of the if branch. Check here https://www.nushell.sh/book/thinking_in_nushell.html#think-of-nushell-as-a-compiled-language and also you might want to use `NU_LIB_DIRS` (see default config: https://github.com/nushell/nushell/blob/a142d1a1926457022d3dce89aa374cf38e967954/docs/sample_config/default_env.nu#L50)

**Comment by skelly37** [comment]
@EdenEast 
As a temporary fix, I can suggest you adding `path expand` to the pipeline, e.g.

`$path | path expand | path exists`

@kubouch 
>Btw, it's not possible to source files from variables

This issue made me think about creating a temporary file, i.e. if *module.nu* exists, append `source module.nu` to the tempfile and in the end source the autogenerated file. Is there a better bypass? Or maybe is this way incorrect?

**Comment by kubouch** [comment]
@skelly37 If you do it within one script, the `source module.nu` line would get appended at the eval time while `source` would read it at parse time and still get the old version, so it wouldn't work. You'd have to do it beforehand but then it's better to use NU_LIB_DIRS.

**Comment by fdncred** [comment]
I think this is fixed by #5886. If not, please reopen.
